src/plugins/intel_gpu/src/runtime/memory_pool.cpp

```diff
@@ -50,24 +50,34 @@ bool memory_pool::has_conflict(const memory_set& a,
 }
 
 void memory_pool::release_memory(memory* mem, const size_t& unique_id, primitive_id prim_id, uint32_t network_id) {
-    // check nonpadded pool first
+    // check non padded pool first
     auto _layout = mem->get_layout();
     auto type = mem->get_allocation_type();
+    const auto _layout_bytes_count = _layout.bytes_count();
 
+    GPU_DEBUG_GET_INSTANCE(debug_config);
     {
-        auto it = _non_padded_pool.lower_bound(_layout.bytes_count());
+        auto it = _non_padded_pool.lower_bound(_layout_bytes_count);
 
         while (it != _non_padded_pool.end()) {
             if (it->second._network_id == network_id &&
                 it->second._type == type &&
                 it->second._memory->get_internal_params().mem == mem->get_internal_params().mem) {
-                auto user_it = it->second._users.find({ unique_id, network_id, prim_id });
+                auto user_it = it->second._users.find({ unique_id, network_id, prim_id, _layout_bytes_count});
```

Comment:
But still, I'd like to suggest to wrap this layout size with DEBUG_CODE to reduce unnecessary query element of the map.

Reply:
Done